src/main/kubeconfig-manager.ts

```diff
@@ -30,57 +30,60 @@ import { LensProxy } from "./lens-proxy";
 import { AppPaths } from "../common/app-paths";
 
 export class KubeconfigManager {
-  protected configDir = AppPaths.get("temp");
-  protected tempFile: string = null;
+  /**
+   * The path to the temp config file
+   *
+   * - if `string` then path
+   * - if `null` then not yet created
+   * - if `undefined` then unlinked by calling `clear()`
+   */
+  protected tempFilePath: string | null | undefined = null;
 
   constructor(protected cluster: Cluster, protected contextHandler: ContextHandler) { }
 
+  /**
+   *
+   * @returns The path to the temporary kubeconfig
+   */
   async getPath(): Promise<string> {
-    if (this.tempFile === undefined) {
+    if (this.tempFilePath === undefined) {
       throw new Error("kubeconfig is already unlinked");
     }
 
-    if (!this.tempFile) {
-      await this.init();
-    }
-
-    // create proxy kubeconfig if it is removed without unlink called
-    if (!(await fs.pathExists(this.tempFile))) {
-      try {
-        this.tempFile = await this.createProxyKubeconfig();
-      } catch (err) {
-        logger.error(`[KUBECONFIG-MANAGER]: Failed to created temp config for auth-proxy`, { err });
-      }
+    if (this.tempFilePath === null || !(await fs.pathExists(this.tempFilePath))) {
+      await this.ensureFile();
     }
 
-    return this.tempFile;
+    return this.tempFilePath;
   }
 
-  async clear() {
-    if (!this.tempFile) {
+  /**
+   * Deletes the temporary kubeconfig file
+   */
+  async clear(): Promise<void> {
+    if (!this.tempFilePath) {
       return;
     }
 
-    logger.info(`[KUBECONFIG-MANAGER]: Deleting temporary kubeconfig: ${this.tempFile}`);
-    await fs.unlink(this.tempFile);
-  }
+    logger.info(`[KUBECONFIG-MANAGER]: Deleting temporary kubeconfig: ${this.tempFilePath}`);
 
-  async unlink() {
-    if (!this.tempFile) {
-      return;
+    try {
+      await fs.unlink(this.tempFilePath);
+    } catch (error) {
+      if (error.code !== "ENOENT") {
+        throw error;
+      }
+    } finally {
+      this.tempFilePath = undefined;
     }
-
-    logger.info(`[KUBECONFIG-MANAGER]: Deleting temporary kubeconfig: ${this.tempFile}`);
-    await fs.unlink(this.tempFile);
-    this.tempFile = undefined;
   }
 
-  protected async init() {
+  protected async ensureFile() {
     try {
       await this.contextHandler.ensureServer();
-      this.tempFile = await this.createProxyKubeconfig();
-    } catch (err) {
-      logger.error(`[KUBECONFIG-MANAGER]: Failed to created temp config for auth-proxy`, err);
+      this.tempFilePath = await this.createProxyKubeconfig();
+    } catch (error) {
+      throw Object.assign(new Error("Failed to creat temp config for auth-proxy"), { cause: error });
```

Comment:
Why it's better than `logger.error`? And can we avoid modifying `Error` object instance?

Reply:
This is more correct because where `ensureFile()` is called, it assumes that `ensureFile()` always succeeds. This way any call to the `getPath()` above will also fail.

I don't think modifying the object like this is bad. Furthermore, `cause` is already natively supported in node 16.9 https://nodejs.org/en/blog/release/v16.9.0/?utm_source=ESnextNews.com&utm_medium=Weekly%20Newsletter&utm_campaign=2021-09-28#error-cause